server/src/main/java/org/opensearch/index/compositeindex/datacube/startree/fileformats/node/FixedLengthStarTreeNode.java

```diff
@@ -0,0 +1,302 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+package org.opensearch.index.compositeindex.datacube.startree.fileformats.node;
+
+import org.apache.lucene.store.RandomAccessInput;
+import org.opensearch.index.compositeindex.datacube.startree.node.StarTreeNode;
+import org.opensearch.index.compositeindex.datacube.startree.node.StarTreeNodeType;
+
+import java.io.IOException;
+import java.io.UncheckedIOException;
+import java.util.Iterator;
+
+/**
+ * Fixed Length implementation of {@link StarTreeNode}.
+ * <p>
+ * This class represents a node in a star tree with a fixed-length serialization format.
+ * It provides efficient storage and retrieval of node information using a RandomAccessInput.
+ * The node structure includes the methods to access all the constructs of InMemoryTreeNode.
+ *
+ * <p>
+ * Key features:
+ * - Fixed-size serialization for each node, allowing for efficient random access
+ * - Binary search capability for finding child nodes
+ * - Support for star nodes, null nodes and other default nodes
+ * - Iteration over child nodes
+ * <p>
+ * The class uses specific byte offsets for each field in the serialized format,
+ * enabling direct access to node properties without parsing the entire node structure.
+ *
+ * @opensearch.experimental
+ */
+public class FixedLengthStarTreeNode implements StarTreeNode {
```

Comment:
Please add UTs for lines which CodeCov has flagged as not covered. Frr new classes, we should be having maximum coverage.